src/zeusops_bot/reforger_config_gen.py

```diff
@@ -4,13 +4,27 @@
 from pathlib import Path
 
 import jsonpatch
+from pydantic import TypeAdapter
 
 from zeusops_bot import errors
 from zeusops_bot.models import ModDetail
 
+modlist_typeadapter = TypeAdapter(list[ModDetail])
+
 SYMLINK_FILENAME = "current-config.json"
 
 
+class Instance:
+    """A config singleton instance"""
+
+    def __init__(self):
+        """Initialise the config instance"""
+        self.config: "ReforgerConfigGenerator" | None = None
+
+
+instance = Instance()
+
```

Comment:
Mega-eww =D

I'd rather figure out how to pass an extra ReforgerConfigGenerator object to either the `ctx` object or to the  autocomplete func directly as param, so that we can have explicit instances, not global singletons implied.
Time to break out the docs of the dependency injection framework they used for the autocomplete

Reply:
This scheme is more or less stolen from the stdlib `logging.getLogger()` method and the `Manager`/root logger singletons of the logging library, but I agree that it's not pretty in any way. Somehow injecting the config into the context could work as well.